robotpy_build/configs.py

```diff
@@ -1,72 +1,73 @@
 # For validating pyproject.toml
 
-from schematics.models import Model
-from schematics.types import ModelType, BooleanType, StringType, ListType, DictType
+from typing import Dict, List, Optional
 
+from pydantic import BaseModel
 
-class WrapperConfig(Model):
+
+class WrapperConfig(BaseModel):
     """
         Wrapper configurations specified in pyproject.toml
 
         [tool.robotpy-build.wrappers."package-name"]
     """
 
     # List of extra headers to export
-    extra_headers = ListType(StringType, default=[])
+    extra_headers: List[str] = []
 
     # List of robotpy-build library dependencies
     # .. would be nice to auto-infer this from the python install dependencies
-    depends = ListType(StringType, default=[])
+    depends: List[str] = []
 
     #
     # Download settings
     #
 
     # Project name
-    name = StringType(required=True)
+    name: str
 
     # Names of contained shared libraries (in loading order). If empty,
     # set to name
-    libs = ListType(StringType)
+    libs: Optional[List[str]] = None
 
     # Name of artifact to download, if different than name
-    artname = StringType(default="")
+    artname: str = ""
 
     # URL to download
-    baseurl = StringType(required=True)
+    baseurl: str
 
     # Version of artifact to download
-    version = StringType(required=True)
+    version: str
 
     # Library extensions map
-    libexts = DictType(StringType, default={})
+    libexts: Dict[str, str] = {}
```

Comment:
Question, does pydantic cause a copy of this dict to be made, or is it shared across instances?

Reply:
From my testing, pydantic does indeed make copies of the default here. They have examples in the docs with an empty list as a default as well.